OpenRA.Game/Actor.cs

```diff
@@ -454,6 +511,61 @@ public IEnumerable<WPos> GetTargetablePositions()
 			return new[] { CenterPosition };
 		}
 
+		#region Conditions
+
+		void UpdateConditionState(string condition, int token, bool isRevoke)
+		{
+			ConditionState conditionState;
+			if (!conditionStates.TryGetValue(condition, out conditionState))
+				return;
+
+			if (isRevoke)
+				conditionState.Tokens.Remove(token);
+			else
+				conditionState.Tokens.Add(token);
+
+			conditionCache[condition] = conditionState.Tokens.Count;
+
+			// Wait until all traits are created.
+			if (created)
```

Comment:
Its not immediately clear that the changes here and in grant/revoke keep equivalent behaviour to the old system (specifically wrt `conditionCache[condition]`). Can you explain why this changed?

IMO it would also be clearer if we could keep the more explicit wording from the previous comments:
```csharp
// Conditions may be granted or revoked before the state is initialized.
// These notifications will be processed in INotifyCreated.Created.
```

Reply:
I updated the comment. It can't be exactly the same since the `ConditionManager` would act some time in the middle of the `INotifyCreated.Created` calls and the postponed notification can only be done before or after here. I have the notifications occur after the calls. If some trait are dependent on being notified before, another notification phase may be needed. This way all the initial tokens are created before initial evaluation. I won't be surprised if some of the more complex traits acquire bugs from this. However, after cleaning those up, this will be a more stable system.